core/sql/optimizer/ItemExpr.cpp

```diff
@@ -14906,3 +14906,80 @@ NABoolean LOBoper::isCovered
   return FALSE;
 }
 
+// Compute the exprssion at compile time. Assume all operands are constants.
+// Return NULL if the computation fails and CmpCommon::diags() may be side-affected.
+ConstValue* ItemExpr::compute(CollHeap* heap)
```

Comment:
If it were a general routine, though, instead of "decodedKeyBuf", maybe "resultBuf". But it is OK as is I think.